crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -247,6 +311,174 @@ fn check_class_attribute(checker: &mut Checker, attribute_expr: &ExprAttribute)
     }
 }
 
+/// Finds the parameter definition for a given name expression in a function.
+fn find_parameter<'a>(
+    semantic: &'a SemanticModel,
+    name: &'a ExprName,
+) -> Option<AnyParameterRef<'a>> {
+    let binding_id = semantic.only_binding(name)?;
+    let binding = semantic.binding(binding_id);
+    let StmtFunctionDef { parameters, .. } = binding.statement(semantic)?.as_function_def_stmt()?;
+    parameters
+        .iter()
+        .find(|parameter| parameter.name().range() == binding.range())
+}
+
+/// Checks whether an Airflow 3.0–removed context key is used in a function decorated with `@task`.
+///
+/// Specifically, it flags two scenarios for task decorated function:
+/// 1. A removed context variable passed in as a function parameter name (e.g., `execution_date`).
+/// 2. A removed context key accessed via `context.get("...")`.
+///
+/// # Examples
+///
+/// **Removed key used in `context.get(...)`:**
+/// ```python
+/// from airflow.decorators import task
+///
+/// @task
+/// def my_task(**context):
+///     # 'conf' is removed in Airflow 3.0
+///     print(context.get("conf"))
+/// ```
+///
+/// **Removed context variable as a parameter:**
+/// ```python
+/// from airflow.decorators import task
+///
+/// @task
+/// def another_task(execution_date, **kwargs):
+///     # 'execution_date' is removed in Airflow 3.0
+///     pass
+/// ```
+///
+/// **Accessing multiple keys:**
+/// ```python
+/// from airflow.decorators import task
+///
+/// @task
+/// def more_keys(**context):
+///     # 'prev_ds' is also removed in Airflow 3.0
+///     print(context.get("prev_ds"))
+/// ```
+fn check_context_get(checker: &mut Checker, call_expr: &ExprCall) {
+    if !is_taskflow(checker) {
+        return;
+    }
+
+    let Expr::Attribute(ExprAttribute { value, attr, .. }) = &*call_expr.func else {
+        return;
+    };
+
+    let is_named_context = if let Expr::Name(name) = &**value {
+        if let Some(parameter) = find_parameter(checker.semantic(), name) {
+            matches!(parameter.name().as_str(), "context" | "kwargs")
+                || parameter.name().as_str().starts_with("**")
+        } else {
+            false
+        }
+    } else {
+        false
+    };
+
+    let is_assigned_from_gcc =
```

Comment:
what is gcc?

Reply:
this could be a function and could be reused in `check_context_variable`

Reply:
It is `get_current_context`. I have changed it to `is_assigned_from_get_current_context`.

Reply:
I think it is just used in check_removed_context_keys_usage function